File: nearly-spring-api/src/main/java/com/ukrainianboyz/nearly/model/entity/UserDto.java
package com.ukrainianboyz.nearly.model.entity;


import com.ukrainianboyz.nearly.db.entity.DatabaseUser;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
@EqualsAndHashCode
public class UserDto {

    String UserId;

    String UserName;

    String UserBio;

    String Email;

    String ImageUrl;

    String AppUserId;

    public UserDto(DatabaseUser dbUser){
        UserId = dbUser.getUserId();
        UserName = dbUser.getUserName();
        UserBio = dbUser.getUserBio();
        Email = dbUser.getEmail();
        ImageUrl = dbUser.getImageUrl();
        AppUserId = dbUser.getAppUserId();
    }

}

File: nearly-android/app/src/main/java/com/ukrainianboyz/nearly/dto/UserDto.kt
package com.ukrainianboyz.nearly.dto

import com.ukrainianboyz.nearly.entity.User

data class UserDto(val userId: String, val userName: String, val email: String, val userBio: String, val imageUrl: String?, val appUserId: String?)


fun UserDto.toUser(): User{
    return User(userId,userName,email,userBio,imageUrl,appUserId)
}
